internal/rbd/rbd_util.go

```diff
@@ -129,15 +130,31 @@ type rbdSnapshot struct {
 
 var (
 	supportedFeatures = sets.NewString("layering")
-
-	// large interval and timeout, it should be longer than the maximum
-	// time an operation can take (until refcounting of the connections is
-	// available)
-	cpInterval = 15 * time.Minute
-	cpExpiry   = 10 * time.Minute
-	connPool   = util.NewConnPool(cpInterval, cpExpiry)
 )
 
+// Connect an rbdVolume to the Ceph cluster
+func (rv *rbdVolume) Connect(cr *util.Credentials) error {
+	if rv.conn != nil {
+		return nil
+	}
+
+	conn := &util.ClusterConnection{}
+	if err := conn.Connect(rv.Monitors, cr); err != nil {
+		return err
+	}
+
+	rv.conn = conn
+	return nil
+}
+
+// Destroy cleans up the rbdVolume and closes the connection to the Ceph
+// cluster in case one was setup.
+func (rv *rbdVolume) Destroy() {
```

Comment:
Disconnect rather than Destroy? Destroy sounds like it might apply to the rbd volume rather than just the cluster connection. To me this is no biggie and could probably be done in a follow up.

Reply:
`Disconnect()` is also not really correct, as the connection is kept open in the util.ConnPool (ok, maybe the comment is not correct either, but implementation details...). I do not like `Destroy()` much either, but it feels a little more suitable.